src/plat/qemu-arm-virt/config.cmake

```diff
@@ -125,9 +125,7 @@ if(KernelPlatformQEMUArmVirt)
             if(NOT DEFINED QEMU_MACHINE)
                 set(QEMU_MACHINE "virt")
                 list(APPEND QEMU_MACHINE "secure=off")
-                if(KernelArmHypervisorSupport OR KernelSel4ArchArmHyp)
-                    list(APPEND QEMU_MACHINE "virtualization=on")
-                endif()
+                list(APPEND QEMU_MACHINE "virtualization=on")
```

Comment:
This now always switches virtualisation on. Do we want that? In fact, I don't understand how this interacts with SMP at all.

Reply:
As the commit message says:

> We always use the qemu EL2 implementation, as this allows the elfloader
to boot secondary cores using SMC (the only method the elfloader
supports). Note this is fine even if CONFIG_ARM_HYPERVISOR_SUPPORT is
not set as the elfloader will drop from EL2 to EL1 before entering the
seL4 kernel.

However, to me it sounds like Elfloader may need to learn to use `HVC` instead of `SMC` when there is no EL3.

This commit seems to be a work around for Elfloader's limitations. Why can't you just enable `KernelArmHypervisorSupport` when using QEMU manually?

Reply:
Yeah this method isn't the best, pasting Gerwin's comment on the other PR
> We do want to test simulated kernels on platforms that don't support virtualisation

I think it'll be better to just implement PSCI secondary core booting via HVC in the elfloader. I'll try and find some time to do this.